lib/spack/spack/test/versions.py

```diff
@@ -808,3 +801,165 @@ def test_git_versions_without_explicit_reference(
+    assert_ver_gt("1.2", "=1.2")
+
+    # StandardVersion / ClosedOpenRange (at different lo version)
+    assert_ver_lt("=1.2", "1.3")
+    assert_ver_gt("1.3", "=1.2")
+    assert_ver_lt("1.2", "=1.3")
+    assert_ver_gt("=1.3", "1.2")
+
+
+def test_git_version_accessors():
+    """Test whether iteration, indexing, slicing, dotted, dashed, and underscored works for
+    GitVersion."""
+    v = GitVersion("my_branch=1.2-3")
+    assert [x for x in v] == [1, 2, 3]
+    assert v[0] == 1
+    assert v[1] == 2
+    assert v[2] == 3
+    assert v[0:2] == Version("1.2")
+    assert v[0:10] == Version("1.2.3")
+    assert str(v.dotted) == "1.2.3"
+    assert str(v.dashed) == "1-2-3"
+    assert str(v.underscored) == "1_2_3"
+    assert v.up_to(1) == Version("1")
+    assert v.up_to(2) == Version("1.2")
+    assert len(v) == 3
+    assert not v.isdevelop()
+    assert GitVersion("my_branch=develop").isdevelop()
+
+
+def test_boolness_of_versions():
+    # We do implement __len__, but at the end of the day versions are used as elements in
+    # the first place, not as lists of version components. So VersionList(...).concrete
+    # should be truthy even when there are no version components.
+    assert bool(Version("1.2"))
+    assert bool(Version("1.2").up_to(0))
+
+    # bool(GitVersion) shouldn't trigger a ref lookup.
+    assert bool(GitVersion("a" * 40))
+
+
+def test_version_list_normalization():
+    # Git versions and ordinary versions can live together in a VersionList
+    assert len(VersionList(["=1.2", "ref=1.2"])) == 2
+
+    # But when a range is added, the only disjoint bit is the range.
+    assert VersionList(["=1.2", "ref=1.2", "ref=1.3", "1.2:1.3"]) == VersionList(["1.2:1.3"])
+
+
+@pytest.mark.parametrize("version", ["=1.2", "git.ref=1.2", "1.2"])
+def test_version_comparison_with_list_fails(version):
+    vlist = VersionList(["=1.3"])
+
+    with pytest.raises(TypeError):
+        version < vlist
+
+    with pytest.raises(TypeError):
+        vlist < version
+
+    with pytest.raises(TypeError):
+        version <= vlist
+
+    with pytest.raises(TypeError):
+        vlist <= version
+
+    with pytest.raises(TypeError):
+        version >= vlist
+
+    with pytest.raises(TypeError):
+        vlist >= version
+
+    with pytest.raises(TypeError):
+        version > vlist
+
+    with pytest.raises(TypeError):
+        vlist > version
+
+
+def test_inclusion_upperbound():
+    is_specific = spack.spec.Spec("x@=1.2")
+    is_range = spack.spec.Spec("x@1.2")
+    upperbound = spack.spec.Spec("x@:1.2.0")
+
+    # The exact version is included in the range
+    assert is_specific.satisfies(upperbound)
+
+    # But the range 1.2:1.2 is not, since it includes for example 1.2.1
+    assert not is_range.satisfies(upperbound)
+
+    # They do intersect of course.
+    assert is_specific.intersects(upperbound) and is_range.intersects(upperbound)
+
+
+@pytest.mark.skipif(sys.platform == "win32", reason="Not supported on Windows (yet)")
```

Comment:
why no windows?

Reply:
It failed, but forgot why. All other `mock_git_version_info` are marked skipif windows too